test/ReadsUtils_server_test.py

```diff
@@ -102,35 +127,35 @@ def delete_shock_node(cls, node_id):
                         allow_redirects=True)
         print('Deleted shock node ' + node_id)
 
+
     @classmethod
     def upload_file_to_shock(cls, file_path):
         """
-        Use HTTP multi-part POST to save a file to a SHOCK instance.
+        Use DataFileUtil.file_to_shock() save a file to a SHOCK instance.
         """
 
-        header = dict()
-        header["Authorization"] = "Oauth {0}".format(cls.token)
-
         if file_path is None:
             raise Exception("No file given for upload to SHOCK!")
 
-        with open(os.path.abspath(file_path), 'rb') as dataFile:
-            files = {'upload': dataFile}
-            print('POSTing data')
-            response = requests.post(
-                cls.shockURL + '/node', headers=header, files=files,
-                stream=True, allow_redirects=True)
-            print('got response')
-
-        if not response.ok:
-            response.raise_for_status()
+        # copy file to where DFU can see it (can only see scratch)
+        src_file_basename = os.path.basename(file_path)
+        shared_file_path = os.path.join(cls.scratch, src_file_basename)
+        shutil.copy2(file_path, shared_file_path)
 
-        result = response.json()
+        # Upload files to shock
+        try:
+            shock_info = cls.dfu.file_to_shock({
+                'file_path': shared_file_path,
+                'make_handle': 1
+            })
+        except Exception as e:
+            raise ValueError('Unable to store ' + file_path + str(e))
 
-        if result['error']:
-            raise Exception(result['error'][0])
-        else:
-            return result["data"]
+        # remember shock info
+        if not hasattr(cls, 'shock_ids'):
+            cls.shock_ids = []
```

Comment:
How is `shock_ids` used?

Reply:
This is a legit question, if Gavin can't understand what this test is doing, then others are likely to as well.